fee-monitor/src/DynamicChecker.ts

```diff
@@ -91,12 +96,209 @@ export class DynamicChecker {
+        const parentBlockTermIndicator = Math.floor(
+            this.parentBlockTimestamp / dynamicParams.termSeconds,
+        );
+        this.parentBlockTimestamp = currentTimestamp;
+
+        if (this.init) {
+            this.parentBlockTimestamp = currentTimestamp;
+            this.parentTermLastBlockNumber = blockNumber - 1;
+            this.init = false;
+        }
+
+        this.handleTransactions(block.transactions, dynamicParams);
+
+        const author = block.author;
+        const weights = await getWeights(blockNumber);
+        const validators = await getValidators(blockNumber);
+        await this.recordContribution(block, validators);
+        distributeDynamic(
+            this.tracerForImmedateSettle,
+            this.currentTermData.intermediateReward,
+            author,
+            weights,
+        );
+
+        const immediateSettleMoment: SettleMoment = {
+            tag: "block",
+            value: blockNumber,
+        };
+
+        if (currentBlockTermIndicator !== parentBlockTermIndicator) {
+            const blockRewardSettleMoment: SettleMoment = {
+                tag: "term",
+                value: blockNumber,
+            };
+            const totalReducedReward = await this.applyPenalty(
+                this.parentTermValidators,
+                blockNumber,
+            );
+            this.applyAdditionalRewards(this.parentTermValidators, totalReducedReward);
+            this.settleNominationDeposit(blockNumber, validators);
+            this.checkForValidators(blockNumber, blockRewardSettleMoment);
+            this.checkForStakeHolders(blockNumber, immediateSettleMoment, true);
+            this.finalizeTerm(blockNumber, validators);
+        } else {
+            this.checkForStakeHolders(blockNumber, immediateSettleMoment, false);
+        }
+    }
+
+
+    private async checkForValidators(currentBlockNumber: number, settleMoment: SettleMoment) {
+        const errorsOfValidators: AggregatedErrors = {
+            server: SERVER,
+            moment: settleMoment,
+            errors: [] as any[],
+        };
+
+        for (const validator of this.parentTermValidators) {
+            const validatorBefore = await sdk.rpc.chain.getBalance(
+                validator,
+                currentBlockNumber - 1,
+            );
+            const validatorAfter = await sdk.rpc.chain.getBalance(validator, currentBlockNumber);
+            const settleStakeHolderReward = this.tracerForImmedateSettle.adjust(
+                validator,
+                validatorBefore,
+            );
 
-    private finalizeTerm(
+            if (
+                !this.prevTermData.intermediateReward
+                    .adjust(validator, settleStakeHolderReward)
+                    .eq(validatorAfter)
+            ) {
+                const error = {
+                    type: "validator",
+                    account: validator.value,
+                    expected: this.tracerForImmedateSettle
```

Comment:
Why duplicate this instead of just `settleStakeHolderReward`? And also, I think it should it be `this.prevTermData.intermediateReward.adjust(validator, settleStakeHolderReward)`

Reply:
Checked